### ISSUE TITLE
expect_lint_free() incompatible with dev testthat

### ISSUE THREAD

**Issue by krlmlr** [jimhester/lintr#137]
Need to use expect_true() instead of expectation(); see also https://github.com/hadley/testthat/pull/407.

CC @hadley

**Comment by jimhester** [comment]
Is there no API for creating user defined expectations in the new testthat implementation?

`testthat::expect()` is not exported. If it were I could use it to fix these errors pretty trivially (https://github.com/jimhester/lintr/pull/141/files)

**Comment by jimhester** [comment]
As you said using `expect_true()` also will work, althought I guess the traceback won't be quite as nice as using `expect()`. https://github.com/jimhester/lintr/pull/141/commits/d72ab0a900c6694b7061784ceba6e9eb4800c406)

**Comment by hadley** [comment]
Maybe we should (re-)expose `expectation()`

**Comment by masalmon** [comment]
With the new testthat version on CRAN expect_lint_free() is incompatible with *not dev* testthat too. The error is (although I guess it's nothing new in this issue)

```r
1. Error: Package Style (@test-lintr.R#9) --------------------------------------
'arg' must be NULL or a character vector
1: lintr::expect_lint_free() at C:\Users\msalmon\ammon/tests/testthat/test-lintr.R:9
2: testthat::expectation(!has_lints, paste(sep = "\n", "Not lint free", lint_output), 
       "lint free")
3: match.arg(type, c("success", "failure", "error", "skip", "warning"))
4: stop("'arg' must be NULL or a character vector")
```

**Comment by jimhester** [comment]
Fixed and on CRAN now.
